axelrod/deterministic_cache.py

```diff
@@ -46,6 +46,24 @@ def __init__(self, file_name=None):
         if file_name is not None:
             self.load(file_name)
 
+    def _key_transform(self, key):
+        """
+        Parameters
+        ----------
+        key: tuple
+            A 3-tuple: (player instance, player instance, match length)
+        """
+        return key[0].name, key[1].name, key[2]
```

Comment:
This is assuming that `key[0]` and `key[1]` are instance of the Player class. I have modified the valid key check to check that that's indeed the case.